### PR TITLE
clang-tidy: fix more checks

### PR DESCRIPTION
**Pull request by scribam**

### CODE REVIEW

File: core/hw/naomi/m4cartridge.h [hunk 1]
```diff
@@ -42,7 +42,7 @@ class M4Cartridge: public NaomiCartridge {
 	bool Read(u32 offset, u32 size, void *dst) override;
 	bool Write(u32 offset, u32 size, u32 data) override;
 
-	void* GetDmaPtr(u32 &size) override;
+	void* GetDmaPtr(u32 &limit) override;
```
Comment (flyinghead) [inline core/hw/naomi/m4cartridge.h:45]: I would prefer to use `size` for consistency with most other cart types. It's also more meaningful for users of the interface.

File: core/hw/sh4/sh4_if.h [hunk 2]
```diff
@@ -345,7 +345,7 @@ s32 rcb_noffs(T* ptr)
 #define Sh4cntx (sh4rcb.cntx)
 
 //Get an interface to sh4 interpreter
-void Get_Sh4Interpreter(sh4_if* cpu);
-void Get_Sh4Recompiler(sh4_if* cpu);
+void Get_Sh4Interpreter(sh4_if* rv);
+void Get_Sh4Recompiler(sh4_if* rv);
```
Comment (flyinghead) [inline core/hw/sh4/sh4_if.h:349]: `rv` probably isn't the best name choice here.

File: core/rend/gles/gles.h [hunk 3]
```diff
@@ -144,7 +144,7 @@ void SetupMatrices(float dc_width, float dc_height,
 				   float &ds2s_offs_x, glm::mat4& normal_mat, glm::mat4& scissor_mat);
 
 text_info raw_GetTexture(TSP tsp, TCW tcw);
-void SetCull(u32 CullMode);
+void SetCull(u32 CulliMode);
```
Comment (flyinghead) [inline core/rend/gles/gles.h:147]: the implementation parameter name should be changed instead

File: core/cfg/cfg.h [hunk 4]
```diff
@@ -6,7 +6,7 @@ bool cfgOpen();
 s32 cfgLoadInt(const std::string& section, const std::string& key, s32 def);
 void cfgSaveInt(const std::string& section, const std::string& key, s32 value);
 std::string cfgLoadStr(const std::string& section, const std::string& key, const std::string& def);
-void cfgSaveStr(const std::string& section, const std::string& key, const std::string& value);
+void cfgSaveStr(const std::string& section, const std::string& key, const std::string& string);
```
Comment (flyinghead) [inline core/cfg/cfg.h:9]: `value` is consistent with other similar functions in this file

File: core/hw/sh4/modules/mmu.h [hunk 5]
```diff
@@ -59,8 +59,8 @@ static INLINE bool mmu_enabled()
 }
 
 template<bool internal = false>
-u32 mmu_full_lookup(u32 va, const TLB_Entry **entry, u32& rv);
-u32 mmu_instruction_lookup(u32 va, const TLB_Entry **entry, u32& rv);
+u32 mmu_full_lookup(u32 va, const TLB_Entry **tlb_entry_ret, u32& rv);
+u32 mmu_instruction_lookup(u32 va, const TLB_Entry **tlb_entry_ret, u32& rv);
```
Comment (flyinghead) [inline core/hw/sh4/modules/mmu.h:63]: I would prefer to keep `entry` here. The implementation name doesn't help readability.

**Review by flyinghead — CHANGES_REQUESTED** [review]
Some of the function parameter name changes aren't ideal. However most are fine and some are clearly needed.

File: core/hw/sh4/modules/mmu.h (hunk 5 above)
Comment (scribam) [reply core/hw/sh4/modules/mmu.h]: I reverted these changes as there is a local variable named `entry` in the implementations

**Pull request merged**